packages/ts/generator-plugin-signals/src/index.ts

```diff
@@ -0,0 +1,76 @@
+import Plugin from '@vaadin/hilla-generator-core/Plugin.js';
+import type SharedStorage from '@vaadin/hilla-generator-core/SharedStorage.js';
+import SignalProcessor, { type MethodInfo } from './SignalProcessor.js';
+
+export type PathSignalType = Readonly<{
+  path: string;
+  signalType: string;
+}>;
+
+const SIGNAL_CLASSES = ['#/components/schemas/com.vaadin.hilla.signals.NumberSignal'];
+
+function extractEndpointMethodsWithSignalsAsReturnType(storage: SharedStorage): PathSignalType[] {
+  const pathSignalTypes: PathSignalType[] = [];
+  Object.entries(storage.api.paths).forEach(([path, pathObject]) => {
```

Comment:
I guess, you could replace it with `map` with `filter` approach or `reduce` to simplify it